src/Microsoft.Health.Dicom.SqlServer/Features/Schema/SqlSchemaVersionResolver.cs

```diff
@@ -0,0 +1,59 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Data.Common;
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+using Microsoft.Health.Dicom.SqlServer.Features.Common;
+
+namespace Microsoft.Health.Dicom.SqlServer.Features.Schema
+{
+    /// <summary>
+    /// Represents an <see cref="ISchemaVersionResolver"/> that determines the version present in a SQL server.
+    /// </summary>
+    public class SqlSchemaVersionResolver : ISchemaVersionResolver
+    {
+        private readonly IDbConnectionFactory _dbConnectionFactory;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SqlSchemaVersionResolver"/> class.
+        /// </summary>
+        /// <param name="dbConnectionFactory">A factory for creating SQL connections.</param>
+        /// <exception cref="ArgumentNullException">
+        /// <paramref name="dbConnectionFactory"/> is <see langword="null"/>.
+        /// </exception>
+        public SqlSchemaVersionResolver(IDbConnectionFactory dbConnectionFactory)
+        {
+            EnsureArg.IsNotNull(dbConnectionFactory, nameof(dbConnectionFactory));
+            _dbConnectionFactory = dbConnectionFactory;
+        }
+
+        /// <summary>
+        /// Asynchronously fetches the current version from the SQL database.
+        /// </summary>
+        /// <param name="cancellationToken">
+        /// The token to monitor for cancellation requests. The default value is <see cref="CancellationToken.None"/>.
+        /// </param>
+        /// <returns>
+        /// A task representing the asychronous operation. The value of its <see cref="Task{TResult}.Result"/>
+        /// property contains the current version in the SQL database.
+        /// </returns>
+        /// <exception cref="OperationCanceledException">The <paramref name="cancellationToken"/> was canceled.</exception>
+        public async Task<SchemaVersion> GetCurrentVersionAsync(CancellationToken cancellationToken = default)
```

Comment:
I found there are multiple implementation for getting current version in shared component, we may want to use one simple approach in the future. 
*  https://github.com/microsoft/healthcare-shared-components/blob/af69cf5660398ffe2980dd31cca6443e1ee7d433/src/Microsoft.Health.SqlServer/Features/Schema/Manager/SchemaManagerDataStore.cs#L108
* https://github.com/microsoft/healthcare-shared-components/blob/af69cf5660398ffe2980dd31cca6443e1ee7d433/src/Microsoft.Health.SqlServer/Features/Schema/SchemaInitializer.cs#L102